tests/connector_test.go

```diff
@@ -124,3 +74,110 @@ func RunStep(t *testing.T, connector deployer.Connector, moduleName string) (str
 
 	return executionResult.OutputID, executionResult.OutputData, executionResult.Error
 }
+
+// This test ensures that this deployer can create and execute
+// connectors concurrently, and that those connectors can deploy
+// plugins concurrently that create side-effects local to their
+// filesystem, and one connector can pull multiple python modules
+func TestDeployConcurrent_ConnectorsAndPluginsWithDifferentModules(t *testing.T) {
+	type TestModule struct {
+		location string
+		stepID   string
+		input    map[string]any
+	}
+	testModules := map[string]TestModule{
+		"fio": {
+			stepID:   "workload",
+			location: "arcaflow-plugin-fio@git+https://github.com/arcalot/arcaflow-plugin-fio.git@de07b3e48cefdaa084eb0445616abc2d13670191",
+			input: map[string]any{
+				"name":    "poisson-rate-submit",
+				"cleanup": "true",
+				"params": map[string]any{
+					"size":           "900KiB",
+					"readwrite":      "randrw",
+					"ioengine":       "sync",
+					"iodepth":        32,
+					"io_submit_mode": "inline",
+					"rate_iops":      50,
+					"rate_process":   "poisson",
+					"buffered":       0,
+				},
+			},
+		},
+		"template": {
+			stepID:   "hello-world",
+			location: "arcaflow-plugin-template-python@git+https://github.com/arcalot/arcaflow-plugin-template-python.git@52d1a9559c60a615dbd97114572f16d70fa30b1b",
+			input: map[string]any{
+				"name": "arca lot",
+			},
+		},
+		"wait": {
+			stepID:   "wait",
+			location: "arcaflow-plugin-wait@git+https://github.com/arcalot/arcaflow-plugin-wait.git",
+			input: map[string]any{
+				"seconds": "0.5",
+			},
+		},
+	}
+
+	rootDir := "/tmp/multi-module"
+	serializedConfig := map[string]any{
+		"workdir":          rootDir,
+		"modulePullPolicy": "Always",
+	}
+
+	assert.NoError(t, os.MkdirAll(rootDir, os.ModePerm))
+
+	factory := pythondeployer.NewFactory()
+	deployerSchema := factory.ConfigurationSchema()
+	unserializedConfig, err := deployerSchema.UnserializeType(serializedConfig)
+	assert.NoError(t, err)
+
+	pythonPath, err := getPythonPath()
+	assert.NoError(t, err)
+	unserializedConfig.PythonPath = pythonPath
+
+	// Choose how many connectors and plugins to make
+	const n_connectors = 4
+	const n_plugin_copies = 10
+	wg := &sync.WaitGroup{}
+	wg.Add(n_connectors * len(testModules) * n_plugin_copies)
```

Comment:
I think line 143 would be better without the ampersand; line 144 will still work even if `wg` is the object and not a pointer to it.